src/libsystemd/sd-device/device-monitor.c

```diff
@@ -91,18 +91,9 @@ int device_monitor_allow_unicast_sender(sd_device_monitor *m, sd_device_monitor
 }
 
 _public_ int sd_device_monitor_set_receive_buffer_size(sd_device_monitor *m, size_t size) {
-        int r, n = (int) size;
-
         assert_return(m, -EINVAL);
-        assert_return((size_t) n == size, -EINVAL);
-
-        if (setsockopt_int(m->sock, SOL_SOCKET, SO_RCVBUFFORCE, n) < 0) {
-                r = setsockopt_int(m->sock, SOL_SOCKET, SO_RCVBUF, n);
-                if (r < 0)
-                        return r;
-        }
 
-        return 0;
+        return fd_inc_rcvbuf(m->sock, size);
```

Comment:
so, hmm, is this right? fd_inc_recvbuf() only *increases* the socket buffer, while the pre-existing code might also lower it if the caller asks for it. Is this an intentional change? I guess in most cases this is called to increase the buffer, but do we want to encode that it can only be used for that?